main.py

```diff
@@ -119,11 +156,37 @@ def main():
         ((650, 100), (650, 150)),  # Dead-end vertical path
         ((100, 450), (350, 450)),  # Dead-end horizontal path
     ]
+
     # Define node properties
     nodes = [end for _, end in paths]
     node_size = 10
     node_color = GREEN
-    node_problems = {tuple(node): random.choice(problems) for node in nodes}
+
```

Comment:
This is impressive work, Good job. 

The questions however that you put here are not in any way related to White Hat Hacking. So that is an issue that needs to be addressed. The theme of the game is to have white hat hacker game scenarios and I do not see how finding sum of two integers or factorial of a number is in any way related to White Hat Hacking. 

So maybe incorporate questions for white hat hacking. :)

Reply:
okay ill try to incoporate some white hat hacking questoin and change the logic to reflect that.